aw_watcher_window/macos.swift

```diff
@@ -85,13 +72,45 @@ struct Heartbeat: Codable {
   var data: NetworkMessage
 }
 
+enum HeartbeatError: Error {
+  case error(msg: String)
+}
+
 struct Bucket: Codable {
   var client: String
   var type: String
   var hostname: String
 }
 
-// Placeholder values, set in start()
+// there's no builtin logging library on macos which has levels & hits stdout, so we build our own simple one
+// there a complex open source one, but it makes it harder to compile this simple one-file swift application
+
+func logTimestamp() -> String {
+  let now = Date()
+  let formatter = DateFormatter()
```

Comment:
Maybe extract this out so that you don't create a formatter every time you want to format a date?

Reply:
Good call, I'll make that change.